features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/BitmapPoolHelper.kt

```diff
@@ -8,12 +8,13 @@ package com.datadog.android.sessionreplay.internal.recorder.resources
 
 import android.graphics.Bitmap
 import com.datadog.android.sessionreplay.internal.utils.InvocationUtils
+import com.datadog.android.sessionreplay.internal.utils.safeConfig
 
 internal class BitmapPoolHelper(
     private val invocationUtils: InvocationUtils = InvocationUtils()
 ) {
     internal fun generateKey(bitmap: Bitmap) =
-        generateKey(bitmap.width, bitmap.height, bitmap.config)
+        generateKey(bitmap.width, bitmap.height, bitmap.safeConfig())
```

Comment:
So how it was working in the past? we now know, that annotation wasn't correct and `Bitmap.getConfig` may return `null`. It should crash runtime, because `generateKey` doesn't expect `null` value for the config parameter.

Also do we need `safeConfig` method? Maybe in some code paths we can handle `null` returned case without fallback value.

Reply:
yes...in case it will return null it will crash. I've never seen this bug reported yet so I suspect this happens quite rarely. I am not sure that this will require a hotfix. @xgouchet @jonathanmos ?

Reply:
the reason why I introduce the `safeConfig` is just to be re - used also in some tests where I had to replace it and maybe for later...looks cleaner this way.

Reply:
The getConfig method is annotated NonNull for some reason. I haven't seen any issues with this returning null, but if it can return it then we do indeed need a safe method.

Reply:
@jonathanmos It was annotated as `NonNull` by mistake, see https://cs.android.com/android/_/android/platform/frameworks/base/+/a4c2f9badb21180282596ac9730d4777f65e4b4b

Reply:
The question was more if we can leave it like this until the next release where the safe method will be in place. You said you did not see any issue so far so I guess this is very rare when it returns null. We could leave it like this without a hotfix.

Reply:
> [!WARNING]
> I'm not at all convinced we need a `safeCongif` here as we lose the information. Any local log or telemetry that could be sent from the generated key will hide the fact that the bitmap configuration was `null`. 
> In the current case, I'd prefer to have a `generateKey` allowing a nullable config instead

Reply:
I didn't see any telemetry there, in any case the Config in this case is just a variable in the key generation so with our without it `null` will have more or less same changes of collisions for 2 elements with `Bitmap.config` null so I am ok with any approach. @jonathanmos , @0xnm what do you guys think ?